lib/tools/system-calls.js

```diff
@@ -153,34 +153,38 @@ systemCallMethods.getBinaryFromPath = async function getBinaryFromPath (binaryNa
  */
 systemCallMethods.getConnectedDevices = async function getConnectedDevices () {
   log.debug('Getting connected devices...');
+  let stdout;
   try {
-    let {stdout} = await exec(this.executable.path, this.executable.defaultArgs.concat(['devices']));
-    // expecting adb devices to return output as
-    // List of devices attached
-    // emulator-5554	device
-    let startingIndex = stdout.indexOf('List of devices');
-    if (startingIndex === -1) {
-      throw new Error(`Unexpected output while trying to get devices. output was: ${stdout}`);
-    }
-    // slicing ouput we care about.
-    stdout = stdout.slice(startingIndex);
-    let devices = [];
-    for (let line of stdout.split('\n')) {
-      if (line.trim() !== '' &&
-          line.indexOf('List of devices') === -1 &&
-          line.indexOf('adb server') === -1 &&
-          line.indexOf('* daemon') === -1 &&
-          line.indexOf('offline') === -1) {
-        let lineInfo = line.split('\t');
-        // state is either "device" or "offline", afaict
-        devices.push({udid: lineInfo[0], state: lineInfo[1]});
-      }
-    }
-    log.debug(`${devices.length} device(s) connected`);
-    return devices;
+    ({stdout} = await exec(this.executable.path, [...this.executable.defaultArgs, 'devices']));
   } catch (e) {
     throw new Error(`Error while getting connected devices. Original error: ${e.message}`);
   }
+  const listHeader = 'List of devices';
+  // expecting adb devices to return output as
+  // List of devices attached
+  // emulator-5554	device
+  const startingIndex = stdout.indexOf(listHeader);
+  if (startingIndex < 0) {
+    throw new Error(`Unexpected output while trying to get devices. Output was: ${stdout}`);
+  }
+  // slicing ouput we care about
+  stdout = stdout.slice(startingIndex);
+  const excludedLines = [listHeader, 'adb server', '* daemon', 'offline'];
+  const devices = stdout.split('\n')
+    .map((line) => line.trim())
```

Comment:
this line can also just be `.map(_.trim)`

Reply:
changed